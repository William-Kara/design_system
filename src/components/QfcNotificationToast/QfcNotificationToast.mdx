import { ArgTypes, Meta } from '@storybook/blocks';
import QfcNotificationToast from '@/components/QfcNotificationToast/QfcNotificationToast.vue';

<Meta title="Modules/Notifications/NotificationToast" />

# Definition & Usage

## Props

The following props are available for the QfcNotificationToast component:

<ArgTypes of={QfcNotificationToast} />

To use QfcNotificationToast, style needs to be imported separately.

```sass
@import '@website/design-system-vuejs/dist/styles/components/QfcBox.scss';
@import '@website/design-system-vuejs/dist/styles/components/QfcNotificationToast.scss';
```

```html
<template>
  <QfcNotificationToast :title="title" :description="description" :type="type" />
</template>
```

- `inline` - Boolean - If true, the notification will be displayed inline is the content size allow it. Default: `true`
