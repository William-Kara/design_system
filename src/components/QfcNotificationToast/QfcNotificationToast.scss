.qfc-notification-toast {
  display: flex;
  align-items: center;
  justify-content: space-between;

  &__content {
    display: flex;
    gap: $q-generic-space-3;
    align-items: center;

    &__texts {
      display: flex;
      gap: 0 $q-generic-space-3;
      font-size: $fs-m;
      flex-wrap: wrap;

      .texts__title {
        font-weight: $bold;
      }
      .texts__description {
        font-weight: $medium;
      }
    }

    &__close-button:hover {
      cursor: pointer;
    }
  }
}

.--color-info {
  color: $info;
}
.--color-success {
  color: $success;
}
.--color-warning {
  color: $warning;
}
.--color-danger {
  color: $danger;
}
.--color-block {
  width: 100%;
}
