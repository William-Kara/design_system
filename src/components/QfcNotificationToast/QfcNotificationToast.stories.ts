import NotificationToast from './QfcNotificationToast.vue';
import type { Meta, StoryObj } from '@storybook/vue3';

const meta: Meta<typeof NotificationToast> = {
  component: NotificationToast,
  title: 'Modules/Notifications/NotificationToast',
  argTypes: {
    type: {
      options: ['success', 'danger', 'warning', 'info'],
    },
    title: {
      control: 'text',
    },
    description: {
      control: 'text',
    },
    inline: {
      control: 'boolean',
    },
  },
};
export default meta;

type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    type: 'info',
    title: 'Title exemple',
    description: 'Lorem exemple',
    inline: true,
  },
};

export const WithLongText: Story = {
  args: {
    ...Primary.args,
    title: 'Title exemple with long text',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec euismod, nisl eget lacinia lacinia, nunc nunc aliquam nunc, vitae aliquam nunc nunc eget nunc. Donec euismod, nisl eget lacinia lacinia, nunc nunc aliquam nunc, vitae aliquam nunc nunc eget nunc.',
  },
};
