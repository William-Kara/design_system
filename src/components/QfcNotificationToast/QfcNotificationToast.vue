<template>
  <QfcNotificationBox :type="type">
    <div class="qfc-notification-toast">
      <div class="qfc-notification-toast__content">
        <span class="material-icons" :class="iconColorStyle">{{ icon }}</span>

        <div class="qfc-notification-toast__content__texts">
          <span class="texts__title" :class="blockStyle">{{ title }}</span>
          <span class="texts__description">{{ description }}</span>
        </div>
      </div>

      <span
        class="qfc-notification-toast__content__close-button material-icons pl-3"
        @click="emitClose"
        >close</span
      >
    </div>
  </QfcNotificationBox>
</template>

<script lang="ts" setup>
import QfcNotificationBox from '@/components/QfcNotificationBox/QfcNotificationBox.vue';
import { computed } from 'vue';

type NotificationType = 'success' | 'danger' | 'warning' | 'info';
interface Props {
  type: NotificationType;
  title: string;
  description: string;
  inline: boolean;
}
const props = withDefaults(defineProps<Props>(), {
  type: 'info',
  inline: true,
});

const iconColorStyle = computed(() => `--color-${props.type}`);
const iconMap = {
  success: 'check_circle',
  danger: 'error',
  warning: 'warning',
  info: 'info',
};
const icon = computed(() => {
  return iconMap[props.type] || 'info';
});

const emits = defineEmits(['close']);
const emitClose = () => {
  emits('close');
};
const blockStyle = computed(() => (props.inline ? '' : '--block'));
</script>

<style lang="scss" scoped src="./QfcNotificationToast.scss" />
