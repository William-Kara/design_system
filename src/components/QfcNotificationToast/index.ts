import QfcNotificationToast from './QfcNotificationToast.vue';

export { QfcNotificationToast };
export default QfcNotificationToast;
