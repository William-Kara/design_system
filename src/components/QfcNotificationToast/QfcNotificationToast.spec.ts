import { it, describe, beforeEach, expect } from 'vitest';

import { shallowMount } from '@vue/test-utils';
import QfcNotificationToast from './QfcNotificationToast.vue';

let wrapper: any = null;

const propsValues = {
  type: 'info',
  title: 'Title exemple',
  description: 'Lorem exemple',
  inline: true,
};

beforeEach(() => {
  wrapper = shallowMount(QfcNotificationToast, {
    propsData: propsValues,
  });
});

describe('QfcNotificationToast Component', () => {
  it('should mount properly', () => {
    expect(wrapper).toBeTruthy();
  });

  it('should have correct props', () => {
    expect(wrapper.props()).toEqual(propsValues);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
