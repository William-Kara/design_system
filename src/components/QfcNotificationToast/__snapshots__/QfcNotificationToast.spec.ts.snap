// Vitest Snapshot v1

exports[`QfcNotificationToast Component > should render correctly 1`] = `"<qfc-notification-box-stub data-v-a30cdd17=\\"\\" type=\\"info\\"></qfc-notification-box-stub>"`;
